### PR TITLE
utils/normalizers: normalize_journal_title proper search

### PR DESCRIPTION
**Pull request by chris-asl**
* Update journals mapping with lowercase analyzer.
* Add `lowercase_titles` field for searching the abbreviated journal
  name.

Signed-off-by: Chris Aslanoglou <chris.aslanoglou@gmail.com>
Co-Authored-By: George Lignos <glignos93@gmail.com>

Addresses #2120 (1st comment)

### CODE REVIEW

File: tests/integration/utils/test_normalizers.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
```
Comment (jacquerie) [inline tests/integration/utils/test_normalizers.py]: Wrong license... where did you even find this one?!

File: inspirehep/utils/normalizers.py [hunk 2]
```diff
@@ -22,19 +22,21 @@
...
+        lowercase_titles=journal_title.lower()
     ).execute()
 
     if hits:
         try:
             normalized_journal_title = hits[0].short_titles[0].title
-        except (AttributeError, IndexError):
-            pass
+        except (AttributeError, IndexError) as e:
+            current_app.logger.debug("Failed to access normalized journal title: {}".format(e))
```
Comment (jacquerie) [inline inspirehep/utils/normalizers.py]: This is not the right way to write logs. See: https://github.com/inspirehep/inspire-next/issues/1371

File: tests/integration/utils/test_normalizers.py (hunk 1 above)
Comment (chris-asl) [inline tests/integration/utils/test_normalizers.py]: Sorry for that, was from the invenio-search `api` file. Will update.

File: inspirehep/utils/normalizers.py (hunk 2 above)
Comment (chris-asl) [inline inspirehep/utils/normalizers.py]: Is there a guidelines document so that I can add this link? 
I think it would be really helpful for others.

File: inspirehep/modules/records/mappings/records/journals.json [hunk 3]
```diff
@@ -175,6 +182,7 @@
                             "type": "string"
                         },
                         "title": {
+                            "copy_to": "lowercase_titles",
```
Comment (jmartinm) [inline inspirehep/modules/records/mappings/records/journals.json]: The field seems to be called `lowercase_journal_titles`?

File: inspirehep/utils/normalizers.py [hunk 4]
```diff
@@ -22,19 +22,22 @@
 
 from __future__ import absolute_import, division, print_function
 
+import logging
```
Comment (jmartinm) [inline inspirehep/utils/normalizers.py]: Was doing some checks about this, and it is better to use:

```python
from flask import current_app

current_app.logger.debug()
```

File: inspirehep/utils/normalizers.py [hunk 5]
```diff
@@ -22,19 +22,22 @@
...
 from inspirehep.modules.search.api import JournalsSearch
 
 
 def normalize_journal_title(journal_title):
+    logger = logging.getLogger(__name__)
     normalized_journal_title = journal_title
     hits = JournalsSearch().query(
         'match',
-        title_variants__title__lowercased=journal_title
+        lowercase_titles=journal_title.lower()
```
Comment (jmartinm) [inline inspirehep/utils/normalizers.py]: I am wondering if this matters, since the analyzer will do the lowercasing.

File: inspirehep/utils/normalizers.py [hunk 6]
```diff
@@ -22,19 +22,22 @@
...
     ).execute()
 
     if hits:
         try:
             normalized_journal_title = hits[0].short_titles[0].title
-        except (AttributeError, IndexError):
-            pass
+        except (AttributeError, IndexError) as e:
+            logger.debug("Failed to access normalized journal title: %s", e)
```
Comment (jmartinm) [inline inspirehep/utils/normalizers.py]: Not sure how useful this message will be.

Could it be better:


```python
logger.debug("Failed to normalize journal title in document: %s", repr(hits[0]))
```

File: tests/integration/utils/test_normalizers.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+from inspirehep.utils.normalizers import normalize_journal_title
+
+
+def test_normalize_journal_title(app):
+    journal_title = 'Physical Review'
+    abbreviated_journal_title = 'Phys.Rev.'
+
+    with app.test_client():
+        normalized_journal_title = normalize_journal_title(journal_title)
```
Comment (jmartinm) [inline tests/integration/utils/test_normalizers.py]: Why was this needed?

File: tests/integration/utils/test_normalizers.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+from __future__ import absolute_import, division, print_function
+
+from inspirehep.utils.normalizers import normalize_journal_title
+
+
+def test_normalize_journal_title(app):
+    journal_title = 'Physical Review'
+    abbreviated_journal_title = 'Phys.Rev.'
+
+    with app.test_client():
```
Comment (jmartinm) [inline tests/integration/utils/test_normalizers.py]: Why was this needed?

File: inspirehep/modules/records/mappings/records/journals.json (hunk 3 above)
Comment (chris-asl) [inline inspirehep/modules/records/mappings/records/journals.json]: Left there during renaming, will update now.

File: inspirehep/utils/normalizers.py (hunk 4 above)
Comment (chris-asl) [inline inspirehep/utils/normalizers.py]: This change happened due to [this comment](https://github.com/inspirehep/inspire-next/pull/2299#pullrequestreview-35949298). 

Which of these two methods for logging, do you guys think I should use?

File: tests/integration/utils/test_normalizers.py (hunk 8 above)
Comment (chris-asl) [inline tests/integration/utils/test_normalizers.py]: When we tested this with George, there was an error (regarding context) that sadly I am no able to re-produce locally now. I'll remove it.

**Comment by chris-asl** [comment]
Actually, as we discussed with @jacquerie, due to the journal suggester PR, [here](https://github.com/inspirehep/inspire-next/pull/2311), I think the normalization (which comes with an extra query to ElasticSearch) isn't needed anymore. 

The integration of the journal suggester in the journals typeahead module, currently sets the value of the text field to the abbreviated title, as was requested by issue #2120. Thus, the normalized value is already there. 

Should I proceed with adding the changes from the later review or simply removing this normalizer?

**Comment by jmartinm** [comment]
> Actually, as we discussed with @jacquerie, due to the journal suggester PR, here, I think the normalization (which comes with an extra query to ElasticSearch) isn't needed anymore.

We will better keep both since even though the autocomplete puts the short title in the input, a user might not select from the autocomplete (or when we import from Crossref we don't do the conversion client side).

I will take over this PR and complete it.

**Comment by jmartinm** [comment]
Continued in https://github.com/inspirehep/inspire-next/pull/2392
